server/routes/caseNotes/caseNotesController.ts

```diff
@@ -14,7 +22,26 @@ export default class CaseNotesController {
       size: 5,
     }
     const caseNotesPage = await this.interventionsService.getCaseNotes(accessToken, req.params.id, paginationQuery)
-    const presenter = new CaseNotesPresenter(caseNotesPage)
+    const uniqueUsers = caseNotesPage.content
+      .map(caseNote => caseNote.sentBy)
+      // filter out duplicate username
+      .filter((user, index, self) => index === self.findIndex(otherUser => otherUser.username === user.username))
+    const userDetails: { username: string; officer: undefined | DeliusUser | AuthUserDetails }[] = await Promise.all(
```

Comment:
why don't we just make this thing return a set of single strings which are either the senders name or the username otherwise?

then the consumer of this code doesn't have to know about undefined `officer`s

Reply:
oh is it more like a map of username -> display name? 

in that case why not make it a map?

`const displayNames: Map<string, string> = await Promise.all( .... )`

Reply:
I thought {username: string, userDetails: unknown} was more appropriate because username is defined with a key.

A map is easier to work with so I've changed to that.